src/graphics/dom.js

```diff
@@ -115,12 +100,15 @@ Crafty.c("DOM", {
      * @param elem - HTML element that will replace the dynamically created one
      *
      * Pass a DOM element to use rather than one created. Will set `._element` to this value. Removes the old element.
+     * 
+     * Will reattach the entity to the current draw layer
      */
     DOM: function (elem) {
         if (elem && elem.nodeType) {
-            this.undraw();
+            var layer = this._drawLayer;
+            this._detachFromLayer();
             this._element = elem;
-            this._element.style.position = 'absolute';
+            this.attachToLayer(layer);
```

Comment:
Good catch, I was probably seeing all elements in the same DOM layer during testing.